course_catalog/api.py

```diff
@@ -526,3 +530,192 @@ def parse_bootcamp_json_data(bootcamp_data, force_overwrite=False):
+        key=lambda x: int(x.key.split("/")[-1].split(".")[0]),
+    ):
+        loaded_raw_jsons_for_course.append(
+            safe_load_json(get_s3_object_and_read(obj), obj.key)
+        )
+
+    log.info("Parsing for %s...", course_prefix)
+    # pass course contents into parser
+    parser = OCWParser("", "", loaded_raw_jsons_for_course)
+    course_json = parser.get_master_json()
+    course_json["uid"] = uid
+    course_json["course_id"] = "{}.{}".format(
+        course_json.get("department_number"), course_json.get("master_course_number")
+    )
+
+    # if course run synced before, update existing Course instance
+    courserun_instance = CourseRun.objects.filter(course_run_id=uid).first()
+
+    # Make sure that the data we are syncing is newer than what we already have
+    if (
+        courserun_instance
+        and last_modified <= courserun_instance.last_modified
+        and not force_overwrite
+    ):
+        log.info("Already synced. No changes found for %s", course_prefix)
+        return uid, is_published
+
+    course_instance = Course.objects.filter(course_id=course_json["course_id"]).first()
+    if upload_to_s3 and is_published:
+        # Upload all course media to S3 before serializing course to ensure the existence of links
+        parser.setup_s3_uploading(
+            settings.OCW_LEARNING_COURSE_BUCKET_NAME,
+            settings.OCW_LEARNING_COURSE_ACCESS_KEY,
+            settings.OCW_LEARNING_COURSE_SECRET_ACCESS_KEY,
+            # course_prefix now has trailing slash so [-2] below is the last
+            # actual element and [-1] is an empty string
+            course_prefix.split("/")[-2],
+        )
+        if settings.OCW_UPLOAD_IMAGE_ONLY:
+            parser.upload_course_image()
+        else:
+            parser.upload_all_media_to_s3(upload_master_json=True)
+
+    log.info("Digesting %s...", course_prefix)
+    digest_ocw_course(
+        parser.get_master_json(),
+        last_modified,
+        course_instance,
+        is_published,
+        course_prefix,
+    )
+    return uid, is_published
+
+
+def sync_ocw_courses(*, force_overwrite, upload_to_s3):
+    """
+    Sync OCW courses to the database
+
+    Args:
+        force_overwrite (bool): A boolean value to force the incoming course data to overwrite existing data
+        upload_to_s3 (bool): If True, upload course media to S3
+
+    Returns:
+        dict: A mapping of CourseRun.course_run_id to its is_published value
+    """
+    raw_data_bucket = boto3.resource(
+        "s3",
+        aws_access_key_id=settings.OCW_CONTENT_ACCESS_KEY,
+        aws_secret_access_key=settings.OCW_CONTENT_SECRET_ACCESS_KEY,
+    ).Bucket(name=settings.OCW_CONTENT_BUCKET_NAME)
+
+    # get all the courses prefixes we care about
+    ocw_courses = generate_course_prefix_list(raw_data_bucket)
+
+    # loop over each course
+    uids = {}
+    for course_prefix in sorted(ocw_courses):
+        try:
```

Comment:
just a tip, to make manual testing faster, I temporarily enclosed the following with this `if` statement before running:
`if course_prefix.startswith("PROD/18/18.06"):`

Reply:
Ah, I did something similar myself